File: examples/undocumented/csharp_modular/kernel_log_modular.cs
using System;

public class kernel_log_modular {
	public static void Main() {
		modshogun.init_shogun_with_defaults();
		double degree = 2.0;

		double[,] traindata_real = Load.load_numbers("../data/fm_train_real.dat");
		double[,] testdata_real = Load.load_numbers("../data/fm_test_real.dat");

		RealFeatures feats_train = new RealFeatures(traindata_real);
		RealFeatures feats_test = new RealFeatures(testdata_real);

		EuclideanDistance distance = new EuclideanDistance(feats_train, feats_train);

		WaveKernel kernel = new WaveKernel(feats_train, feats_test, degree, distance);

		double[,] km_train = kernel.get_kernel_matrix();
		kernel.init(feats_train, feats_test);
		double[,] km_test = kernel.get_kernel_matrix();

		foreach (double item in km_train)
		      Console.Write(item);

		foreach (double item in km_test)
		      Console.Write(item);

		modshogun.exit_shogun();
	}
}

File: examples/undocumented/python_modular/kernel_log_modular.py
#!/usr/bin/env python
traindat = '../data/fm_train_real.dat'
testdat = '../data/fm_test_real.dat'

parameter_list=[[traindat,testdat, 2.0],[traindat,testdat, 3.0]]

def kernel_log_modular (train_fname=traindat,test_fname=testdat, degree=2.0):
	from modshogun import RealFeatures, LogKernel, EuclideanDistance, CSVFile

	feats_train=RealFeatures(CSVFile(train_fname))
	feats_test=RealFeatures(CSVFile(test_fname))

	distance=EuclideanDistance(feats_train, feats_train)

	kernel=LogKernel(feats_train, feats_train, degree, distance)
	km_train=kernel.get_kernel_matrix()

	kernel.init(feats_train, feats_test)
	km_test=kernel.get_kernel_matrix()
	return km_train,km_test,kernel


if __name__=='__main__':
	print('Log')
	kernel_log_modular(*parameter_list[0])
